packages/dev/test/RegistrationManager.test.ts

```diff
@@ -45,23 +45,29 @@ contract('RegistrationManager', function (accounts) {
   // When running server before staking/funding it, or when balance gets too low
   describe('multi-step server initialization', function () {
     // TODO: It does not make sense for the '_worker' method to expose the reason it does not register
+    //       It could expose the types of transactions it has broadcast to simplify logging, testing & debugging.
     //       This means these 2 tests cannot check what they used to and require refactoring.
-    it('should wait for balance', async function () {
+    it('should wait for balance before setting owner on the StakeManager', async function () {
       let latestBlock = await env.web3.eth.getBlock('latest')
       let transactionHashes = await relayServer._worker(latestBlock.number)
       assert.equal(transactionHashes.length, 0)
       const expectedBalance = env.web3.utils.toWei('2', 'ether')
-      assert.notEqual((await relayServer.getManagerBalance()).cmp(toBN(expectedBalance)), 0)
+      let managerBalance = await relayServer.getManagerBalance()
+      assert.notEqual(managerBalance.cmp(toBN(expectedBalance)), 0)
       await env.web3.eth.sendTransaction({
         to: relayServer.managerAddress,
         from: relayOwner,
         value: expectedBalance
       })
       latestBlock = await env.web3.eth.getBlock('latest')
+      managerBalance = await relayServer.getManagerBalance()
+      assert.equal(managerBalance.cmp(toBN(expectedBalance)), 0, 'should have balance now')
       transactionHashes = await relayServer._worker(latestBlock.number)
-      assert.equal(transactionHashes.length, 0)
+      assert.equal(transactionHashes.length, 1, 'should only set owner')
+      const tx = await web3.eth.getTransaction(transactionHashes[0])
+      assert.equal(tx.to, env.stakeManager.address, 'should only set owner')
+      assert.equal(tx.input.indexOf('0xfece3dd4'), 0, 'should only set owner')
```

Comment:
yuck. why not test emitted event?

Reply:
This is a bit tricky with the way Transaction Manager just broadcasts transactions, so I would have to manually decode logs after getting raw transaction from the node. 
Not sure if this is worth it for a single test